packages/repository/src/decorators/relation.decorator.ts

```diff
@@ -78,36 +132,38 @@ export function hasOne(definition?: Object) {
  * @returns {(target:any, key:string)}
  */
 export function hasMany<T extends typeof Entity>(
-  targetModel: T,
+  targetModel: T | TypeResolver<T>,
   definition?: Partial<HasManyDefinition>,
 ) {
   // todo(shimks): extract out common logic (such as @property.array) to
   // @relation
   return function(target: Object, key: string) {
     property.array(targetModel)(target, key);
 
-    const defaultFkName = camelCase(target.constructor.name + '_id');
-    const hasKeyTo = definition && definition.keyTo;
-    const hasDefaultFkProperty =
-      targetModel.definition &&
-      targetModel.definition.properties &&
-      targetModel.definition.properties[defaultFkName];
-    if (!(hasKeyTo || hasDefaultFkProperty)) {
-      // note(shimks): should we also check for the existence of explicitly
-      // given foreign key name on the juggler definition?
-      throw new Error(
-        `foreign key ${defaultFkName} not found on ${
-          targetModel.name
-        } model's juggler definition`,
-      );
+    const meta: Partial<HasManyDefinition> = {target: targetModel};
+
+    if (!isTypeResolver(targetModel)) {
+      const defaultFkName = camelCase(target.constructor.name + '_id');
```

Comment:
There is a way to autofill `keyTo`, it just has to be done during runtime when `createHasManyRepositoryFactory` or the belongsTo counterpart is run. I agree that this block of code should be removed entirely since it's not possible to determine `keyTo` at the decorator level when there is a cyclic dependency.